### ISSUE TITLE
Accessibility: Table's column header does not announce name, state, and role

### ISSUE THREAD

**Issue by rezinak** [highcharts/highcharts#18517]
#### Expected behaviour
Custom controls provide proper textual name, role, and state information. Properly structured native controls provide this information automatically. For custom controls, developers must explicitly add this information by using ARIA attributes.

A control's name must represent the general purpose of the control and include any visible labeling text. It can be set through a variety of methods. Custom controls typically use the aria-label attribute or the aria-labelledby attribute to supply the name.

A control's role indicates the type of control, such as "checkbox" or "button". For custom controls, the role must be set with the role attribute. The ARIA specification only permits certain values for this attribute. Additionally, the ARIA in HTML specification does prevent the use of certain roles on some elements. For a list of HTML elements and their permitted roles, see the ARIA in HTML specification: https://www.w3.org/TR/html-aria/#docconformance

A control's state indicates several possible pieces of information. Common states include expanded, current, selected, checked, and disabled. Most are indicated with similarly named ARIA attributes. Certain states can only be used with certain roles. For a list of roles and their allowed states, see the ARIA specification: https://www.w3.org/TR/wai-aria/#role_definitions

#### Actual behaviour
Each of the sortable column headers do not announce appropriate name, role and state information.
Screen reader users will have difficulty determining the purpose and state of these controls.

#### Live demo with steps to reproduce
https://www.idc.com/QA_test_section/accessibility-remediation/highcharts-demo1 view data table

#### Product version
Highcharts JS v10.3.3

#### Affected browser(s)
Chrome and Mozilla, haven't looked into the others

**Comment by oysteinmoseng** [comment]
Internal note: Table sorting is undercommunicated in general, we should also consider if we should indicate better that this is a feature to all users (sort icon?). For screen readers, either add a sort-button in each column header, or make each entire column header a button. Also see the [WAI authoring practices](https://w3c.github.io/aria-practices/examples/table/sortable-table.html) on this, illustrating `aria-sort`.
